<?xml version="1.0" encoding="UTF-8"?>
<launch>

	<arg name="veh" default="SQ"/>
	<arg name="num" default="01"/> <!-- use this for snap_sim (if quad is SQ01s then num is 01) -->

	<arg name="x" default="0.0"/>
	<arg name="y" default="0.0"/>
	<arg name="z" default="0.8" />
	<arg name="yaw" default="0.0" />

	<arg name="goal_topic" default="goal"/>

	<!-- Start mader commands -->
	<node ns="$(arg veh)$(arg num)s" name="mader_commands" pkg="mader" type="mader_commands.py" output="screen"/>

	<!-- Start Perfect tracker or snap_sim
	<include file="$(find mader)/launch/perfect_tracker_or_sim.launch">
  		<arg name="quad" value="$(arg veh)$(arg num)s"/>
  		<arg name="num" value="$(arg num)"/>
  		<arg name="gazebo" value="false"/>
        <arg name="x" value="$(arg x)"/>
        <arg name="y" value="$(arg y)"/>
        <arg name="z" value="$(arg z)"/>
        <arg name="yaw" default="$(arg yaw)"/>
        <arg name="perfect_tracker" default="false"/>
 	 </include>
 	-->

 	<!-- mader -->
	<node ns="$(arg veh)$(arg num)s" name="mader" pkg="mader" type="mader_node" output="screen" clear_params="false" >
		<rosparam file="$(find mader)/param/mader.yaml" command="load"/>
	        <!-- Subscriptions -->
			<remap from="~state" to="state"/>
			<remap from="~who_plans" to="who_plans"/>
			<remap from="~term_goal" to="term_goal" />

			<!-- Publications -->
			<remap from="~traj" to="traj"/>
			<remap from="~goal" to="$(arg goal_topic)" />
			<remap from="~setpoint" to="setpoint" />
	</node>

	<group if="$(eval num == '01')">
	    <node ns="$(arg veh)$(arg num)s" name="mader" pkg="mader" type="mader_node" output="screen" clear_params="false" >
			<rosparam file="$(find mader)/param/mader1.yaml" command="load"/>
		        <!-- Subscriptions -->
				<remap from="~state" to="state"/>
				<remap from="~who_plans" to="who_plans"/>
				<remap from="~term_goal" to="term_goal" />

				<!-- Publications -->
				<remap from="~traj" to="traj"/>
				<remap from="~goal" to="$(arg goal_topic)" />
				<remap from="~setpoint" to="setpoint" />
		</node>
	</group>
	<group if="$(eval num == '02')">
	    <node ns="$(arg veh)$(arg num)s" name="mader" pkg="mader" type="mader_node" output="screen" clear_params="false" >
			<rosparam file="$(find mader)/param/mader2.yaml" command="load"/>
		        <!-- Subscriptions -->
				<remap from="~state" to="state"/>
				<remap from="~who_plans" to="who_plans"/>
				<remap from="~term_goal" to="term_goal" />

				<!-- Publications -->
				<remap from="~traj" to="traj"/>
				<remap from="~goal" to="$(arg goal_topic)" />
				<remap from="~setpoint" to="setpoint" />
		</node>
	</group>
	<group if="$(eval num == '03')">
	    <node ns="$(arg veh)$(arg num)s" name="mader" pkg="mader" type="mader_node" output="screen" clear_params="false" >
			<rosparam file="$(find mader)/param/mader3.yaml" command="load"/>
		        <!-- Subscriptions -->
				<remap from="~state" to="state"/>
				<remap from="~who_plans" to="who_plans"/>
				<remap from="~term_goal" to="term_goal" />

				<!-- Publications -->
				<remap from="~traj" to="traj"/>
				<remap from="~goal" to="$(arg goal_topic)" />
				<remap from="~setpoint" to="setpoint" />
		</node>
	</group>
	<group if="$(eval num == '04')">
	    <node ns="$(arg veh)$(arg num)s" name="mader" pkg="mader" type="mader_node" output="screen" clear_params="false" >
			<rosparam file="$(find mader)/param/mader4.yaml" command="load"/>
		        <!-- Subscriptions -->
				<remap from="~state" to="state"/>
				<remap from="~who_plans" to="who_plans"/>
				<remap from="~term_goal" to="term_goal" />

				<!-- Publications -->
				<remap from="~traj" to="traj"/>
				<remap from="~goal" to="$(arg goal_topic)" />
				<remap from="~setpoint" to="setpoint" />
		</node>
	</group>
	<group if="$(eval num == '05')">
	    <node ns="$(arg veh)$(arg num)s" name="mader" pkg="mader" type="mader_node" output="screen" clear_params="false" >
			<rosparam file="$(find mader)/param/mader5.yaml" command="load"/>
		        <!-- Subscriptions -->
				<remap from="~state" to="state"/>
				<remap from="~who_plans" to="who_plans"/>
				<remap from="~term_goal" to="term_goal" />

				<!-- Publications -->
				<remap from="~traj" to="traj"/>
				<remap from="~goal" to="$(arg goal_topic)" />
				<remap from="~setpoint" to="setpoint" />
		</node>
	</group>
	<group if="$(eval num == '06')">
	    <node ns="$(arg veh)$(arg num)s" name="mader" pkg="mader" type="mader_node" output="screen" clear_params="false" >
			<rosparam file="$(find mader)/param/mader6.yaml" command="load"/>
		        <!-- Subscriptions -->
				<remap from="~state" to="state"/>
				<remap from="~who_plans" to="who_plans"/>
				<remap from="~term_goal" to="term_goal" />

				<!-- Publications -->
				<remap from="~traj" to="traj"/>
				<remap from="~goal" to="$(arg goal_topic)" />
				<remap from="~setpoint" to="setpoint" />
			</node>
	</group>

</launch>
